common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -319,7 +337,7 @@ def get_html(self):
             )
 
         completion_service = self.runtime.service(self, 'completion')
-        if completion_service:
+        if completion_service and view != PUBLIC_VIEW:
```

Comment:
Instead of this, can you look into not enabling the completion service if the user is anonymous here: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/lms_xblock/runtime.py#L140.

In case the user us signed-in but unenrolled, can you try setting the `effective_user` to `AnonymousUser()` if `self.view is PublicView` here: https://github.com/edx/edx-platform/blob/5f9fe4a529237f348059a77baa7218275fab8145/lms/djangoapps/courseware/views/index.py#L162

Reply:
@symbolist I added the check to disable completion service if user is anonymous at the above mentioned line in runtime.py file and also set the `effective_user` to `AnonymousUser()` in the index.py file. However, while testing I noticed that the completion service was being set before the `effective_user` change (i.e. runtime.py file is executed before the index.py) and so the completion service was being enable for the unenrolled (but signed-in) users. I removed the change from the index.py file and added a check for unenrolled user while setting the completion service. Let me know if that needs to be changed.

Reply:
Hmm. In my testing, `LMSModuleSystem` is initialized inside [`render()`](https://github.com/edx/edx-platform/blob/5f9fe4a529237f348059a77baa7218275fab8145/lms/djangoapps/courseware/views/index.py#L152). From which method do you see it getting called before `_setup_masquerade_for_effective_user()`?

```
__init__ [LmsModuleSystem] runtime.py:140
get_module_system_for_user module_render.py:820
get_module_for_descriptor_internal module_render.py:877
get_module_for_descriptor module_render.py:433
_prefetch_and_bind_course [CoursewareIndex] index.py:331
render [CoursewareIndex] index.py:150
get [CoursewareIndex] index.py:126
```

Reply:
I think I found the reason for that. While testing, I happened to first click on the course which directed to the url `/courses/<course-id>/about` (which initialised `LMSModuleSystem` at https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/module_render.py#L765) and then I changed it to `/courses/<course-id>/course`. 
But if I open `/courses/<course-id>/course` directly (i.e. without opening other urls), it initialises `LMSModuleSystem` from [render()](https://github.com/edx/edx-platform/blob/5f9fe4a529237f348059a77baa7218275fab8145/lms/djangoapps/courseware/views/index.py#L152). Having the check while enabling/disabling the completion service handles both the scenarios. But if that's unnecessary, I can go with the other approach of setting the `effective_user` to `AnonymousUser()`. Let me know your thoughts.

Reply:
Thanks @symbolist ! That makes sense. I've made the required changes and squashed all the commits. Let me know if any other changes are required.